pkgpanda/__init__.py

```diff
@@ -792,10 +797,10 @@ def _get_service_names(_dir):
             if package.username is not None:
                 sysusers.add_user(package.username, package.group)
 
-            # Ensure the state directory in `/var/lib/dcos` exists
+            # Ensure the state directory exists
             # TODO(cmaloney): On upgrade take a snapshot?
             if self.__manage_state_dir:
-                state_dir_path = '/var/lib/dcos/{}'.format(package.name)
+                state_dir_path = self.__state_dir_root + '/' + package.name
```

Comment:
Yeah, that's the big reason that we don't use `os.path.join()` everywhere. I'm sure we could write our own helper function around it that does what we like. Raising an error if `os.path.sep` is in a path segment before we call `os.path.join()` would be a good start.